internal/vm/checker/checker.go

```diff
@@ -27,16 +25,18 @@ type checker struct {
 
 	errors errors.ErrorList
 
-	depth   int
-	tooDeep bool
+	depth             int
+	tooDeep           bool
+	maxRecursionDepth int
+	maxRegexLength    int
 }
 
 // Check performs a semantic check of the astNode, and returns a potentially
 // modified astNode and either a list of errors found, or nil if the program is
 // semantically valid.  At the completion of Check, the symbol table and type
 // annotation are also complete.
-func Check(node ast.Node) (ast.Node, error) {
-	c := &checker{}
+func Check(node ast.Node, maxRegexLength int, maxRecursionDepth int) (ast.Node, error) { // TODO is there a way to inject these integers without adding them as part of this constructor?
```

Comment:
Not sure if this is something you guys do, but I left this todo because I'm not very happy with the changes I had to make here. I want to make this better before merging.  But I can't think of any way to make it better. To any other reviewer, do you have a better idea of how I could inject these integers in?

Reply:
The checker and the rest of the compiler do not support the Options pattern, so this is your only choice right now, so drop the TODO.